### ISSUE TITLE
Problems on atlas

### ISSUE THREAD

**Issue by JairoApa** [realmonster/Gens-Atlas#2]
Can't view atlas, only see the normal game

**Comment by realmonster** [comment]
You need to set path to sonic atlas dll in settings window. It works only with sonic 3 & knuckles.

**Comment by realmonster** [comment]
Also, if you interested in recording into video file, read this thread https://tasvideos.org/Forum/Topics/20802?CurrentPage=2&Highlight=485539#485539
I have also made hack-around changes to this Gens-Atlas to make possible capture, but didn't commit it here. You can find corresponding showcase video in the same thread.

**Comment by JairoApa** [comment]
![Captura de pantalla 2024-09-12 205425](https://github.com/user-attachments/assets/c31fda69-6a0a-41c6-8d69-70ace98951ca)

**Comment by JairoApa** [comment]
![image](https://github.com/user-attachments/assets/1eaa7b71-6c4f-440c-9b00-7674dd15b79a)
![image](https://github.com/user-attachments/assets/1ebf2c95-a468-4ef3-843a-65236f694b9e)
![image](https://github.com/user-attachments/assets/04f88d2f-ada7-4056-b88d-6eaf12e542de)

I have this errors

**Comment by realmonster** [comment]
I see only one obvious mistake:
[this line](https://github.com/realmonster/Gens-Atlas/blob/f947b7bb09ab8689b7b91b34b6fe00c62ee001b0/Sonic3-Atlas/main.cpp#L812)
`	for (int i = links.size()-1; i>=0; --i)`
should be changed into
`	for (int i = (int)links.size()-1; i>=0; --i)`
other than that, I don't know.

Try debug it yourself. On first screen click "retry" as mentioned in the message box. It'll show where exactly it crashed. Similarly, on second screen click "break" as explained in the message box. I don't want to tell instructions how to debug, because it's cumbersome to explain how to do it. Other than this, to debug dll it's better to set current project to project of the dll.

**Comment by JairoApa** [comment]
But the error says that it is a file called "vector" that is in the directory: C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\include\

It is indicating that the vector subscript is out of range.

**Comment by realmonster** [comment]
It can be "out of range" at any part of the code during the execution of the code. So you need to figure out where it goes out of range.

**Comment by JairoApaESP** [comment]
And how do I do that?

**Comment by JairoApa** [comment]
And how can I do that?

**Comment by realmonster** [comment]
I want to stress out **atlas works only with Sonic 3 or Sonic 3 and Knuckles.**
I was testing with Sonic 3 & Knuckles SHA1 hash `cfbf98c36c776677290a872547ac47c53d2761d6`.
And Sonic 3 SHA1 hash `75e9c4705259d84112b3e697a6c00a0813d47d71`.

I was able to successfully build this stuff in Microsoft Visual Studio 2022 Community Edition.

1. Clone or download & unpack source tree from this repository.
2. Open solution file (.sln) in Visual Studio
3. Click with Right Mouse Button on solution. Select "Retarget solution". Select update platform to v145 or something like that (your current platform)
4. Solution->Build. You'll see errors. Solution->Build again. You'll still see errors.
5. Find the error where it say something like "string is not from std". Click on error it will open source file. In the beginning of the file find `#include <vector>` and add new line `#include <string>` after it.
6. Download [glew](https://glew.sourceforge.net/). (Download win32 binaries). Extract into project folder.
7. Right click on gens project. Select "preferences". Linker > Advanced property page. Image Has Safe Exception Handlers property -> set NO. (this should get rid of errors SAFESEH) (It should be /SAFESEH:NO)
8. Right click on gens project. Select "preferences". In section with VC directories paths, add GL directory from glew, into include directories. Add libs directory to libraries directories.
9. Do the same as in (8) with Sonic Atlas project.
10. Build gens and sonic-atlas. This time it should compile fine.
11. Copy glew32.dll into "Output" directory of project.
12. Manually run gens.exe from output directory. **Not from Visual Studio.**
13. In gens in Options->BIOS add path to renderer. (sonic atlas dll from output folder)
14. Reopen gens.
15. Now you should see additional blank window next to gens window -- it's sign that renderer works. Resize window to see atlas.

Steps 8 and 9 should get rid of errors with "can't open gl/glew.h" and glew32.lib not found.

**Comment by JairoApa** [comment]
![image](https://github.com/user-attachments/assets/73dc73fe-fab8-43fb-a96c-e2388eab2dfa)
I have 7 errors on Sonic3-Atlas project

**Comment by realmonster** [comment]
In properties of Sonic Atlas project disable Unicode character set. I don't know in which section of your VS it is. For example Advanced->Character set.

**Comment by JairoApaESP** [comment]
https://github.com/user-attachments/assets/48f766d2-dc97-4fb0-81df-683ee93cb4e6

I have this errors now

**Comment by JairoApa** [comment]
![image](https://github.com/user-attachments/assets/11c8fc5a-3e1d-4e70-ad8b-63094fc7ac14)
![image](https://github.com/user-attachments/assets/78bd3369-29b7-4d57-95a4-8e5a78262266)
![image](https://github.com/user-attachments/assets/8046b2d7-9e1a-4603-9771-cc7f9a288b86)
![image](https://github.com/user-attachments/assets/3261f321-870e-4583-bf30-fa56b4372f27)
I have this errors

**Comment by JairoApa** [comment]
But when enable the unicode character i have 11 errors
![image](https://github.com/user-attachments/assets/4c707369-1fcf-440d-8459-bf9a3bc3603e)
![image](https://github.com/user-attachments/assets/2fba6fd4-d308-4af7-9183-39ae816c44a9)

**Comment by realmonster** [comment]
Unicode should be disabled for Sonic Atlas project.
Regarding linker errors with unresolved symbols, make sure you have in project settings in linker settings in list of libs to link: glew32.lib and opengl32.lib. Make sure in the beginning it doesn't tell you it can't find them.

Regarding to /RTCc, in project settings C/C++ > Code Generation change "Basic Runtime Checks" or "Smaller Type Check". I don't know how exactly change it. Try different options.

**Comment by JairoApa** [comment]
C1189	#error:  /RTCc rejects conformant code, so it is not supported by the C++ Standard Library. Either remove this compiler option, or define _ALLOW_RTCc_IN_STL to suppress this error.	gens	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\include\yvals_core.h	1858		
![image](https://github.com/user-attachments/assets/49bf81a6-8904-41be-b1c0-ebef577dd0ec)

**Comment by realmonster** [comment]
In my previous reply I said you need to disable /RTCc

**Comment by JairoApa** [comment]
No, but is the gens project, the Sonic-Atlas project I have been able to build it 
[Sonic3-Atlas.zip](https://github.com/user-attachments/files/17101205/Sonic3-Atlas.zip)

**Comment by realmonster** [comment]
Anyway I answered what to do to get rid of error from previous screenshot.

**Comment by JairoApa** [comment]
Could you send me a build already done?

**Comment by realmonster** [comment]
[gens-atlas.zip](https://github.com/user-attachments/files/17119271/gens-atlas.zip)

**Comment by JairoApa** [comment]
Thanks, it works but can't turn fullscreen mode, it closes the gens atlas
![image](https://github.com/user-attachments/assets/491f6ffc-bb08-4afe-9830-fb0510cf0e43)
![image](https://github.com/user-attachments/assets/16d37faf-b436-4a02-8115-40f9eabfa5e3)

**Comment by realmonster** [comment]
I don't plan fix it.

**Comment by JairoApa** [comment]
And how do I dump an avi with the atlas?

**Comment by realmonster** [comment]
this build can't do that

**Comment by JairoApa** [comment]
The thing is that in this video that you uploaded https://youtu.be/O37VNoOiFUU in the description I found this link: https://gist.github.com/realmonster/99ee2dbd2c1dc2bdc8136b3826e5b11a/

I downloaded the file and I don't know how to use it.

**Comment by realmonster** [comment]
Use [git apply](https://stackoverflow.com/questions/2249852/how-to-apply-a-patch-generated-with-git-format-patch)
Then use AVI recording as usual.
I recommend to install [Lagarith Lossless Video Codec](https://lags.leetcode.net/codec.html) and encode using it. Output would be large but you can compress it later by ffmpeg for example. Don't forget to enable split AVI 1900 MB for example.
I also recommend apply [these changes](https://github.com/TASEmulators/gens-rerecording/commit/a2425b5244edbd4f345f7462af48fefaac0ea709#diff-41d4202c240cd883e0dc07dc8d146271b23c67045f403dc5dbac564a9ae7a6bdL614) as far as I remember it fixes single frame loss during avi splits.

**Comment by JairoApa** [comment]
Still not recording in atlas, just recording normally

**Comment by realmonster** [comment]
did you rebuild gens?

**Comment by JairoApaESP** [comment]
No, download the gens-atlas.zip file you sent me earlier

**Comment by JairoApa** [comment]
No, download the gens-atlas.zip file you sent me earlier

**Comment by JairoApa** [comment]
In fact in a publication: https://tasvideos.org/Forum/Topics/5904?Highlight=347266&CurrentPage=75&PageSize=25&Sort=CreateTimestamp I found these links could you confirm me if in those you can dump an avi in atlas,when I open the links I get "This content is no longer available"?
![image](https://github.com/user-attachments/assets/9bcc7c76-0353-4299-9679-442ab14d1958)
